### PR TITLE
promutil(engine): extract wrapping factory for DM

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #5493

### What is changed and how it works?

DM needs to attach labels for dumpling/lightning, so split the functionality of old wrappingFactory.

#### Tests 

 - pass old tests

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/pkg/promutil/registry.go [hunk 1]
```diff
@@ -94,3 +94,58 @@ func (r *Registry) Gather() ([]*dto.MetricFamily, error) {
...
+}
+
+func (f *AutoRegisterFactory) NewGaugeVec(opts prometheus.GaugeOpts, labelNames []string) *prometheus.GaugeVec {
+	c := f.inner.NewGaugeVec(opts, labelNames)
+	f.r.MustRegister(f.id, c)
+	return c
+}
+
+func (f *AutoRegisterFactory) NewHistogram(opts prometheus.HistogramOpts) prometheus.Histogram {
+	c := f.inner.NewHistogram(opts)
```
Comment (maxshuang) [inline engine/pkg/promutil/registry.go:142]: No golint error for  the missing of comments?

Comment (lance6716) [reply engine/pkg/promutil/registry.go:142]: verify CI is down last Friday, now it works and I'll fix it.

File: engine/pkg/promutil/registry.go [hunk 2]
```diff
@@ -94,3 +94,58 @@ func (r *Registry) Gather() ([]*dto.MetricFamily, error) {
 	// NOT NEED lock here. prometheus.Registry has thread-safe methods
 	return r.Registry.Gather()
 }
+
+// AutoRegisterFactory uses inner Factory to create metrics and register metrics
+// to Registry with id. Panic if it can't register successfully.
```
Comment (maxshuang) [inline engine/pkg/promutil/registry.go:99]: Seems splitting the registration is more flexible.
Thanks for the Pr.

**Review by amyangfei — COMMENTED** [review]
rest lgtm

**Review by amyangfei — APPROVED** [review]

**Review by liuzix — APPROVED** [review]

**Pull request merged**